src/main.rs

```diff
@@ -484,28 +496,11 @@ fn test_complete_command() {
     );
 }
 
-fn remove_groff_formatting(data: &str) -> String {
-    // TODO Can we remove all of these strings in one go?
-    let mut data = data.to_owned();
-    for marker in &[
-        r"\fI", r"\fP", r"\f1", r"\fB", r"\fR", r"\e", r".BI", r".BR", r"0.5i", r".rb", r"\^",
-        r"{ ", r" }", ".B",
-        ".I",
-        //     The next ones are odd. Putting them into a python file makes my
-        //     python linter warn about anomalous backslash and python2 vs python3
-        //     seems to make no difference
-        //     data = data.replace("\ ","")
-        //     data = data.replace("\-","-")
-        //"\ ",
-        //"\&",
-        //"\f",
-    ] {
-        data = data.replace(marker, "");
-    }
-    // See note above about anomalous backslash
-    //data = data.replace("\-", "-");
-    let data = regex!(r##".PD( \d+)"##).replace_all(&data, "");
-    data.to_string()
+fn remove_groff_formatting(data: &str) -> Cow<'_, str> {
+    // using regex is twice as fast as manual replace
+    let re =
```

Comment:
Are we sure this is strictly equivalent? I _hope_ the original wasn't relying on the explicit ordering, but we would want to verify that. Here's a contrived example of a string that the original would treat differently than a regex replace-all: `r###"\\fIfP"###`. The original will transform that to `""`, but the regex replace all will only do a single pass, which will result in `r###"\fP"###`.

Maybe retain some of the comments from original and a TODO to revisit?

Reply:
Okay, but what I did is to replace all, the original regex wasn't doing that.